hazelcast/src/test/java/com/hazelcast/internal/partition/impl/FrozenPartitionTableTest.java

```diff
@@ -131,10 +132,13 @@ private void partitionTable_isFrozen_whenMemberReJoins_duringClusterStateIs(Clus
         terminateInstance(hz2);
         terminateInstance(hz3);
 
-        hz3 = factory.newHazelcastInstance(hz3Address);
-        final Member newMember3 = getClusterService(hz3).getLocalMember();
+        HazelcastInstance newInstance = factory.newHazelcastInstance(hz3Address);
+        hz3 = newInstance;
+        Member newMember3 = getClusterService(hz3).getLocalMember();
 
         assertClusterSizeEventually(2, hz1, hz3);
+        // ensure partition state is applied on the new member
+        assertTrueEventually(() -> Accessors.isPartitionStateInitialized(newInstance));
```

Comment:
we can't use `waitForClusterSafe` here because the cluster is not safe (it's missing a member)